src/frauddetectionservice/src/main/kotlin/frauddetectionservice/main.kt

```diff
@@ -0,0 +1,44 @@
+package frauddetectionservice
+
+import org.apache.kafka.clients.consumer.ConsumerConfig.*
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.kafka.common.serialization.StringDeserializer
+import org.apache.kafka.common.serialization.ByteArrayDeserializer
+import hipstershop.Demo.*
+import java.time.Duration.ofMillis
+import java.util.Properties
+import kotlin.system.exitProcess
+
+const val topic = "orders"
+const val groupID = "frauddetectionservice"
+
+fun main(args: Array<String>) {
+    val props = Properties()
+    props[KEY_DESERIALIZER_CLASS_CONFIG] = StringDeserializer::class.java.name
+    props[VALUE_DESERIALIZER_CLASS_CONFIG] = ByteArrayDeserializer::class.java.name
+    props[GROUP_ID_CONFIG] = groupID
+    val bootstrapServers = System.getenv("KAFKA_SERVICE_ADDR")
+    if (bootstrapServers == null) {
+        println("KAFKA_SERVICE_ADDR is not supplied")
+        exitProcess(1)
+    }
+    props[BOOTSTRAP_SERVERS_CONFIG] = bootstrapServers
+    val consumer = KafkaConsumer<String, ByteArray>(props).apply {
+        subscribe(listOf(topic))
+    }
+
+    var totalCount = 0L
+
+    consumer.use {
+        while (true) {
+            totalCount = consumer
+                .poll(ofMillis(100))
+                .fold(totalCount) { accumulator, record ->
+                    val newCount = accumulator + 1
+                    val orders = OrderResult.parseFrom(record.value())
+                    println("Consumed record with key ${record.key()} and value $orders, and updated total count to $newCount")
```

Comment:
We should not log the entire message here. Just a key about it (tracking id maybe?) and/or the fact that we read a message.